### ISSUE TITLE
ldap authentification

### ISSUE THREAD

**Issue by jqt-ing** [3liz/lizmap-web-client#780]
Hi,

I have en issue with the ldap module for LIZMAP. I have read [lizmap documentation](https://docs.3liz.com/en/install/linux.html#ldap-authentication)

I have download with the module, and copy the directory "ldapdao" in the path "/var/www/lizmap-web-client-3.1.6/lizmap/lizmap-modules/".

I followed the instructions:
>apt-get install php7.1-ldap
> 
> Declare the ldapdao module into the lizmap/var/config/localconfig.ini.php file
> 
> [modules]
> ldapdao.access=1
> Check the following modules state into the lizmap/var/config/localconfig.ini.php file
> 
> jacl2.access=1
> jauth.access=2
> jauthdb.access=1
> 
> Redefine the path of the auth config into the lizmap/var/config/admin/config.ini.php and lizmap/var/config/index/config.ini.php files
> 
> [coordplugins]
> auth="authldap.coord.ini.php"
> Create a profile like this according to your ldap settings into the lizmap/var/config/profiles.ini.php file
> 
> [ldap:myldapdao]
> hostname=localhost
> port=389
> adminUserDn="cn=admin,ou=admins,dc=acme"
> adminPassword="Sup3rP4ssw0rd"

But when I run `Run php lizmap/install/installer.php`, I have the error:

> PHP Fatal error:  Uncaught Exception: Error in the main configuration. A plugin configuration file doesn't exist -- Configuration file for the coord plugin auth doesn't exist: 'authldap.coord.ini.php' in /var/www/lizmap-web-client-3.1.6/lib/jelix/core/jConfigCompiler.class.php:172
> Stack trace:
> #0 /var/www/lizmap-web-client-3.1.6/lib/jelix/core/jConfigCompiler.class.php(134): jConfigCompiler::checkCoordPluginsPath(Object(stdClass))
> #1 /var/www/lizmap-web-client-3.1.6/lib/jelix/core/jConfigCompiler.class.php(81): jConfigCompiler::prepareConfig(Object(stdClass), true, false, '/index.php')
> #2 /var/www/lizmap-web-client-3.1.6/lib/jelix/installer/jInstallerEntryPoint.class.php(89): jConfigCompiler::read('index/config.in...', true, false, '/index.php')
> #3 /var/www/lizmap-web-client-3.1.6/lib/jelix/installer/jInstaller.class.php(341): jInstallerEntryPoint->__construct(Object(jIniFileModifier), 'index/config.in...', 'index.php', 'classic')
> #4 /var/www/lizmap-web-client-3.1.6/lib/jelix/installer/jInstaller.class.php(298): jInstaller->ge in /var/www/lizmap-web-client-3.1.6/lib/jelix/core/jConfigCompiler.class.php on line 172

what can I check? did I forget something?

Thanks and regards

**Comment by rldhont** [comment]
It's an error in the Documentation.

The step

> Redefine the path of the auth config into the lizmap/var/config/admin/config.ini.php and lizmap/var/config/index/config.ini.php files

has to be done after the `php lizmap/install/installer.php`

Remove
```
[coordplugins]
auth="authldap.coord.ini.php"
```

And launch `php lizmap/install/installer.php`

**Comment by josemvm** [comment]
the documentation has been updated in github
https://github.com/3liz/lizmap-documentation/blob/master/source/install/linux.rst#ldap-authentication

it needs to be uploaded to the documentation page

**Comment by signedav** [comment]
The documentation is correctly updated. Thanks for that.

But I'm stucking at the same point, so I attach on this issue since it's still open. Tell me if I should open an other one, please.

When I try to install like:
```
php ./var/www/websig/lizmap/install/installer.php
```
I receive:
```
Installation start..
[notice] Installation starts for the entry point index
[error] Name is missing in a dependency declaration in module ldapdao comp=ldapdao
[error] Error in dependencies. Installation cancelled.
Installation/upgrade is aborted for the entry point index
Installation ended.
```

**What I did:**
I've Lizmap 3.1.4. I did the following things for the LDAP integration:
1. installed php5-ldap
```
apt-get install php5-ldap
```

2. copied the ldapdao to the lizmap-modules folder
```
docker cp ldapdao-module/ldapdao/ 851363823bbf:/var/www/websig/lizmap/lizmap-modules
```

3. added the section in `/var/www/websig/lizmap/var/config/localconfig.ini.php`
```
[modules]
ldapdao.access=1
```

4.checked ` /var/www/websig/lizmap/var/config/mainconfig.ini.php`
Should look like this
```
jacl2.access=1
jauth.access=2
jauthdb.access=1
```
But looks like this:
```
jacl2.access=1
jauth.access=2
jauthdb.access=2
```
I did not change anything though.

5. I run php `./var/www/websig/lizmap/install/installer.php`
And I receive the mentioned feedback.

Thanks a lot!

**Comment by signedav** [comment]
This problem does only occur on master checkout. With the latest version of ldapdao (2.0.4) there's another message on installation:
```
All modules dependencies are ok
PHP Notice:  Undefined index: ldapdao in /var/www/websig/lizmap/lizmap-modules/ldapdao/install/install.php on line 25
PHP Warning:  SQLite3::exec(): NOT NULL constraint failed: jacl2_rights.id_aclgrp in /var/www/websig/lib/jelix/plugins/db/sqlite3/sqlite3.dbconnection.php on line 120
[error] An error occured during the installation of the module ldapdao: invalid query (NOT NULL constraint failed: jacl2_rights.id_aclgrp (INSERT INTO jacl2_rights (
id_aclsbj,id_aclgrp,id_aclres,canceled
) VALUES (
'auth.user.change.password', NULL, '-', 0
)))
Installation ended.
```

Anyway with 2.0.0 or 2.0.4 the installation succeeds - still I have trouble with the driver #1001

**Comment by laurentj** [comment]
@josemvm doc updated on docs.lizmap.com
@signedav I released a 2.0.5 that fix the issue `Undefined index`

**Comment by josemvm** [comment]
thanks @laurentj :-)
